.editorconfig

```diff
@@ -233,32 +231,34 @@ dotnet_naming_style.first_upper.capitalization = first_word_upper
 dotnet_naming_style.prefix_interface_interface_with_i.capitalization = pascal_case
 dotnet_naming_style.prefix_interface_interface_with_i.required_prefix = I
+# disallowed_by_design_guidelines - use to auto-format items that are disallowed by the design guidelines
+dotnet_naming_style.prefix_with_underscore.capitalization = pascal_case
+dotnet_naming_style.prefix_with_underscore.required_prefix = _
+# disallowed_by_design_guidelines - use to auto-format items that are disallowed by the design guidelines
+dotnet_naming_style.disallowed_by_design_guidelines.capitalization = pascal_case
+dotnet_naming_style.disallowed_by_design_guidelines.required_prefix = ____INVALID____
+dotnet_naming_style.disallowed_by_design_guidelines.required_suffix = ____INVALID____
 
+######################################################################
 # Naming Rules
-# Constant fields must be PascalCase
-dotnet_naming_rule.constant_fields_must_be_pascal_case.severity = warning
-dotnet_naming_rule.constant_fields_must_be_pascal_case.symbols = constant_fields
-dotnet_naming_rule.constant_fields_must_be_pascal_case.style = pascal_case
-# Public, internal and protected readonly fields must be PascalCase
-dotnet_naming_rule.non_private_readonly_fields_must_be_pascal_case.severity = warning
-dotnet_naming_rule.non_private_readonly_fields_must_be_pascal_case.symbols = non_private_readonly_fields
-dotnet_naming_rule.non_private_readonly_fields_must_be_pascal_case.style = pascal_case
-# Static readonly fields must be PascalCase
-dotnet_naming_rule.static_readonly_fields_must_be_pascal_case.severity = warning
-dotnet_naming_rule.static_readonly_fields_must_be_pascal_case.symbols = static_readonly_fields
-dotnet_naming_rule.static_readonly_fields_must_be_pascal_case.style = pascal_case
-# Private readonly fields must be camelCase
-dotnet_naming_rule.private_readonly_fields_must_be_camel_case.severity = warning
-dotnet_naming_rule.private_readonly_fields_must_be_camel_case.symbols = private_readonly_fields
-dotnet_naming_rule.private_readonly_fields_must_be_camel_case.style = camel_case
-# Public and internal fields must be PascalCase
-dotnet_naming_rule.public_internal_fields_must_be_pascal_case.severity = warning
-dotnet_naming_rule.public_internal_fields_must_be_pascal_case.symbols = public_internal_fields
-dotnet_naming_rule.public_internal_fields_must_be_pascal_case.style = pascal_case
-# Private and protected fields must be camelCase
-dotnet_naming_rule.private_protected_fields_must_be_camel_case.severity = warning
-dotnet_naming_rule.private_protected_fields_must_be_camel_case.symbols = private_protected_fields
-dotnet_naming_rule.private_protected_fields_must_be_camel_case.style = camel_case
+######################################################################
+# private fields can be per user preference
+# (including defining additional naming symbols above for more fine-grained control)
+dotnet_naming_rule.private_fields_rule.symbols  = private_fields
```

Comment:
I'm trying to get my head around what changed and how it translates. A few bullet points would be nice.

From what I've gathered, the official naming conventions are quite permissive and less restrictive than my rules which I actually derived from StyleCop. I'm not sure what effect your changes would have, so I'd have to try it and see. Interested to hear your comments.

Reply:
Quick response to provide bullet points, per your request.

### Background:
From https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-naming-conventions:
> Naming conventions should be ordered from most-specific to least-specific
> in the .editorconfig file. The first rule encountered that can be applied
> is the only rule that is applied.

From https://editorconfig.org/#file-location:
> EditorConfig files are read top to bottom and the most recent rules found take precedence.
> Properties from matching EditorConfig sections are applied in the order they were read,
> so properties in closer files take precedence.

https://editorconfig.org/#file-format-details
> EditorConfig files are read top to bottom and the most recent rules found take precedence.

Thus, both of the following are true:
* lines of the file are processed in order (as one would expect)
* for **naming conventions**, only the **_first_** applicable rule is applied


 _See also https://github.com/MicrosoftDocs/visualstudio-docs/issues/1351_, where I ask the official sources about related questions in the multi-file scenario.

Reply:
Hi RehanSaeed,

To get your head around the changes, please look at pull request https://github.com/RehanSaeed/EditorConfig/pull/4, and more specifically the table I created at https://github.com/RehanSaeed/EditorConfig/pull/4/commits/9b7d00267946a90bc1073eeb2f6318c9f8f03f21#diff-1e70daafb475c0ce3fef7d2728279182R282.

This table lists what symbol types are matched by each of the old rules, lists what I believe occurs under the new rules, and finishes with a [table summarizing only the differences](https://github.com/RehanSaeed/EditorConfig/pull/4/commits/9b7d00267946a90bc1073eeb2f6318c9f8f03f21#diff-1e70daafb475c0ce3fef7d2728279182R282).